Game crashes when wanting to enter Map Screen

#### Bug description

Version Label: Stracciatella v0.23.0-git+442b71a1a
Version #:     Build 04.12.02

---

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
|  v0.23.0-git+442b71a1a | JA2 linux version  | not sure | i (belive) yes |

---

JA2S Sci-Fi mode.

After building the git version to verify https://github.com/ja2-stracciatella/ja2-stracciatella/issues/2395 if it is really fixed. It is fixed. Thanks.

To trigger the "Crashes on Using Map Screen" bug over here. Load the GameSave.
The logs:

    2026-07-30T18:35:12.004869497Z [INFO] game/GameLoop.cc: Version #:     Build 04.12.02
    2026-07-30T18:35:12.016035267Z [ERROR] game/Utils/Cinematics.cc: Failed to open 'intro/splashscreen.smk': SGPFile::openInVfs: Vfs_open "intro/splashscreen.smk": entity not found
    2026-07-30T18:35:20.468574512Z [ERROR] sgp/SGP.cc: Game has been terminated due to an unrecoverable error:  Vehicle has no corresponding soldier
    Creating an emergency savegame succeeded (Error.sav).
    Please report this error with a description of the circumstances.  Do not forget to attach the savegame.   (St13runtime_error)
    [1]    10917 IOT instruction  ja

It is sector N6, it has 6 mercs and 1 vehicle (hummer).

[QuickSave.zip](https://github.com/user-attachments/files/30558357/QuickSave.zip)

#### Steps to reproduce
1. Load QuickSave
2. JA2 enters in tactical screen (6 merc's middle of a combat)
3. Use then the "Map Scren" Button to get to the "MAP" overview of the game
4. Game crashes.

This is the game generated error.sav 

[Error.zip](https://github.com/user-attachments/files/30558380/Error.zip)

This is correct. I could narrow it to destroying the tank in the sector. The Hummer is not involved. So the last message as the game crashes might be related to the tank, after it has been destroyed.

[Crash.zip](https://github.com/user-attachments/files/30595305/Crash.zip)

1) Load the Crash.zip Savegame
2) Move Barry 1 step further, so he sees the tank.
3) Now Ingame verify the Map Screen Button is still working properly and does not crash the game
4) Destroy the tank with 1/2 shots using the mortar on mercs MD and Needle.
5) As soon as the tank is destroyed, and one uses the Map Screen Button the game crashes reporting:


       2026-07-31T15:29:49.585580921Z [INFO] stracciatella::logger: Logging to file "/tmp/ja2.log"
       2026-07-31T15:29:49.605347986Z [WARN] stracciatella::mods: Could not read mod manifest for `nightops-maps` at location `"/home/needle/.ja2/mods/nightops-maps/manifest.json"`: No such file or directory (os error 2)
       2026-07-31T15:29:49.67063846Z [INFO] stracciatella::vfs: VFS layer 1: SlfFs { source: DirFsFile { "tilesets.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670663531Z [INFO] stracciatella::vfs: VFS layer 2: SlfFs { source: DirFsFile { "speech.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670677484Z [INFO] stracciatella::vfs: VFS layer 3: SlfFs { source: DirFsFile { "sounds.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.67069197Z [INFO] stracciatella::vfs: VFS layer 4: SlfFs { source: DirFsFile { "radarmaps.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670705286Z [INFO] stracciatella::vfs: VFS layer 5: SlfFs { source: DirFsFile { "npcdata.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670718401Z [INFO] stracciatella::vfs: VFS layer 6: SlfFs { source: DirFsFile { "npc_speech.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670731436Z [INFO] stracciatella::vfs: VFS layer 7: SlfFs { source: DirFsFile { "music.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670744399Z [INFO] stracciatella::vfs: VFS layer 8: SlfFs { source: DirFsFile { "mercedt.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670757732Z [INFO] stracciatella::vfs: VFS layer 9: SlfFs { source: DirFsFile { "maps.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670770735Z [INFO] stracciatella::vfs: VFS layer 10: SlfFs { source: DirFsFile { "loadscreens.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670783931Z [INFO] stracciatella::vfs: VFS layer 11: SlfFs { source: DirFsFile { "laptop.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670797503Z [INFO] stracciatella::vfs: VFS layer 12: SlfFs { source: DirFsFile { "interface.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670810625Z [INFO] stracciatella::vfs: VFS layer 13: SlfFs { source: DirFsFile { "fonts.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670824036Z [INFO] stracciatella::vfs: VFS layer 14: SlfFs { source: DirFsFile { "faces.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670837409Z [INFO] stracciatella::vfs: VFS layer 15: SlfFs { source: DirFsFile { "data.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.67085042Z [INFO] stracciatella::vfs: VFS layer 16: SlfFs { source: DirFsFile { "cursors.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670863546Z [INFO] stracciatella::vfs: VFS layer 17: SlfFs { source: DirFsFile { "binarydata.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.6708767Z [INFO] stracciatella::vfs: VFS layer 18: SlfFs { source: DirFsFile { "bigitems.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670890107Z [INFO] stracciatella::vfs: VFS layer 19: SlfFs { source: DirFsFile { "battlesnds.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670903001Z [INFO] stracciatella::vfs: VFS layer 20: SlfFs { source: DirFsFile { "anims.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670915975Z [INFO] stracciatella::vfs: VFS layer 21: SlfFs { source: DirFsFile { "ambient.slf" in "/opt/ja2/data" } }
       2026-07-31T15:29:49.670928945Z [INFO] stracciatella::vfs: VFS layer 22: DirFs { "/opt/ja2/data" }
       2026-07-31T15:29:49.670943268Z [INFO] stracciatella::vfs: VFS layer 23: DirFs { "/opt/ja2-straciatella.git/externalized" }
       2026-07-31T15:29:49.670958286Z [INFO] stracciatella::vfs: VFS layer 24: DirFs { "/opt/ja2-straciatella.git/mods/from-russia-with-love/data" }
       2026-07-31T15:29:49.670973466Z [INFO] stracciatella::vfs: VFS layer 25: DirFs { "/opt/ja2-straciatella.git/mods/generous-rebels/data" }
       2026-07-31T15:29:49.670988648Z [INFO] stracciatella::vfs: VFS layer 26: DirFs { "/opt/ja2-straciatella.git/mods/imp-quiz-honest-answers/data" }
       2026-07-31T15:29:49.671003878Z [INFO] stracciatella::vfs: VFS layer 27: DirFs { "/home/needle/.ja2/mods/kaerar-gun-pack/Data" }
       2026-07-31T15:29:49.671029713Z [INFO] externalized/DefaultContentManager.cc: JA2 Home Dir:                  '/home/needle/.ja2'
       2026-07-31T15:29:49.671044914Z [INFO] externalized/DefaultContentManager.cc: Root game resources directory: '/opt/ja2'
       2026-07-31T15:29:49.671061192Z [INFO] externalized/DefaultContentManager.cc: Extra data directory:          '/opt/ja2-straciatella.git'
       2026-07-31T15:29:49.671073287Z [INFO] externalized/DefaultContentManager.cc: Saved games directory:         '/home/needle/.ja2/SavedGames'
       2026-07-31T15:29:49.6710852Z [INFO] externalized/DefaultContentManager.cc: Temporary directory:           '/tmp/ja2-stracciatella-SCsAkpxf'
       2026-07-31T15:29:49.671098383Z [INFO] externalized/ModPackContentManager.cc: Enabled mods                    'from-russia-with-love, generous-rebels, imp-quiz-honest-answers, kaerar-gun-pack'
       2026-07-31T15:29:50.441253628Z [INFO] game/GameLoop.cc: Version Label: Stracciatella v0.23.0-git+442b71a1a
       2026-07-31T15:29:50.441301482Z [INFO] game/GameLoop.cc: Version #:     Build 04.12.02
       2026-07-31T15:29:50.454727508Z [ERROR] game/Utils/Cinematics.cc: Failed to open 'intro/splashscreen.smk': SGPFile::openInVfs: Vfs_open "intro/splashscreen.smk": entity not found
       2026-07-31T15:34:49.58561806Z [ERROR] sgp/SGP.cc: Game has been terminated due to an unrecoverable error: Vehicle has no corresponding soldier
       Creating an emergency savegame succeeded (Error.sav).
       Please report this error with a description of the circumstances.  Do not forget to attach the savegame. (St13runtime_error)
       [1]    3160 IOT instruction  ja2
       ja2  97.12s user 5.17s system 34% cpu 5:00.34 total

Attaching the Error.zip 

[Error.zip](https://github.com/user-attachments/files/30595329/Error.zip)

--- a/src/game/Strategic/Map_Screen_Interface_Map.cc
+++ b/src/game/Strategic/Map_Screen_Interface_Map.cc
@@ -722,6 +722,7 @@ static INT32 ShowVehicles(const SGPSector& sSector, INT32 icon_pos)
 		// skip the chopper, it has its own icon and displays in airspace mode
 		if (IsHelicopter(v))                          continue;
 		if (v.sSector != sSector)                     continue;
+		if (v.fDestroyed)                             continue;
 		if (PlayerIDGroupInMotion(v.ubMovementGroup)) continue;
 
 		SOLDIERTYPE const& vs = GetSoldierStructureForVehicle(v);

--- a/src/game/Tactical/Vehicles.cc
+++ b/src/game/Tactical/Vehicles.cc
@@ -118,7 +118,10 @@ INT32 AddVehicleToList(const SGPSector& sMap, const INT16 sGridNo, const UINT8 u
 void RemoveVehicleFromList(VEHICLETYPE& v)
 {
 	v.pMercPath = ClearStrategicPathList(v.pMercPath, 0);
-	v = VEHICLETYPE{};
+	// We cannot really remove a vehicle from the vector because
+	// SOLDIERTYPE::bVehicleID is an index into this vector, so just
+	// mark this entry as invalid.
+	v.fValid = false;
 }
 
 

